### PR TITLE
Tournament API

### PR DESCRIPTION
**Pull request by SarahhhC**
* Send tournament list with career area.
(exclude user with any connection.)
(status 'All' is for 'doesn't matter' option in client.)
* Unit test for tournament added.

### CODE REVIEW

File: src/server/controllers/tournament.controller.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+/*
+ * Methods for getting tournament list.
+ */
+
+const ObjectId = mongoose.Types.ObjectId;
+
+const Match = mongoose.model('match');
+const User = mongoose.model('user');
+
+export function getTournamentList(req, res, next) {
```
Comment (MechanicKim) [inline src/server/controllers/tournament.controller.js]: 1. Prefer default export
2. 'next' is defined but never used

File: src/server/controllers/tournament.controller.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    mentor_id: 1,
+  };
+
+  const localField = {
+    mentee: 'mentee_id',
+    mentor: 'mentor_id',
+  };
+
+  Promise.all([
+    findConnection(matchOptions.MATCH_AS_MENTOR, projectOption, localField.mentee),
```
Comment (MechanicKim) [inline src/server/controllers/tournament.controller.js]: line 33, 34
'findConnection' was used before it was defined

File: src/server/controllers/tournament.controller.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    findConnection(matchOptions.MATCH_AS_MENTEE, projectOption, localField.mentor),
+  ])
+    .then((exceptionList) => {
+      let defaultFindOption = {
```
Comment (MechanicKim) [inline src/server/controllers/tournament.controller.js]: 'defaultFindOption' is never reassigned. Use 'const' instead

File: src/server/controllers/tournament.controller.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        _id: {
+          $ne: req.user._id,
+          $nin: exceptionList[0],
+          $nin: exceptionList[1],
```
Comment (MechanicKim) [inline src/server/controllers/tournament.controller.js]: Duplicate key '$nin'

File: src/server/controllers/tournament.controller.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        },
+        mentorMode: {
+          $ne: false,
+        },
+      };
+
+      if (req.params.area === 'All') {
+        return User.find(defaultFindOption, { password: 0 })
+          .sort({ stamp_login: -1 }).exec();
+      } else {
```
Comment (MechanicKim) [inline src/server/controllers/tournament.controller.js]: Unnecessary 'else' after 'return'

File: src/server/controllers/tournament.controller.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+          'career.area': req.params.area,
+        }, { password: 0 })
+          .sort({ stamp_login: -1 }).exec();
+      }
+    })
+    .then((userList) => {
+      res.status(200).json(userList);
+    })
+    .catch((err) => {
+      res.status(400).json({ err: err });
```
Comment (MechanicKim) [inline src/server/controllers/tournament.controller.js]: Expected property shorthand
err: err ---> err

File: src/server/controllers/tournament.controller.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+    });
+}
+
+function findConnection(matchOption, projectOption, localField) {
+  return Match.aggregate([{
+    $match: matchOption,
+  },
+    {
```
Comment (MechanicKim) [inline src/server/controllers/tournament.controller.js]: line 72, 74, 75, 82 : Expected indentation of 2 spaces but found 4
line 73, 76 : Expected indentation of 4 spaces but found 6

File: src/server/controllers/tournament.controller.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+      $project: projectOption,
+    },
+    {
+      $lookup: {
+        from: 'users',
+        localField: localField,
```
Comment (MechanicKim) [inline src/server/controllers/tournament.controller.js]: Expected property shorthand
localField: localField ---> localField

File: src/server/routes/tournament.route.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
+import express from 'express';
+import jwtUtil from '../utils/jwt.util';
+import * as tournament from '../controllers/tournament.controller';
+
+const apiProtector = jwtUtil.apiProtector;
+const router = express.Router();
+
+//GET method
```
Comment (MechanicKim) [inline src/server/routes/tournament.route.js]: Are commented lines useful?

File: src/server/controllers/tournament.controller.js (hunk 3 above)
Comment (SarahhhC) [inline src/server/controllers/tournament.controller.js]: ok!

File: src/server/controllers/tournament.controller.js (hunk 4 above)
Comment (SarahhhC) [inline src/server/controllers/tournament.controller.js]: This is just for operation! :)

File: src/server/controllers/tournament.controller.js (hunk 6 above)
Comment (SarahhhC) [inline src/server/controllers/tournament.controller.js]: good!

File: src/server/controllers/tournament.controller.js (hunk 8 above)
Comment (SarahhhC) [inline src/server/controllers/tournament.controller.js]: good!

File: src/server/controllers/tournament.controller.js (hunk 5 above)
Comment (SarahhhC) [inline src/server/controllers/tournament.controller.js]: Seems unnecessary!

File: src/server/routes/tournament.route.js (hunk 9 above)
Comment (SarahhhC) [inline src/server/routes/tournament.route.js]: Nope, just for convenience!

File: src/server/controllers/tournament.controller.js (hunk 1 above)
Comment (SarahhhC) [inline src/server/controllers/tournament.controller.js]: I tried to use default but some err occurred!

**Comment by ace0625** [comment]
LGTM

**Pull request merged**